hasher/src/hasher/endpoints.py

```diff
@@ -33,13 +33,15 @@ async def heart_beat() -> str:
     summary="Produce secure hash with optional max output length (2 <= length <= 64)",
 )
 async def hash(message: str, length: int = 64) -> Response:  # noqa: A001
-    output = generate_hash(message, length)
+    hasher = Hasher()
```

Comment:
could we just have a `hasher` object created upon startup of the app and we use that in all of the subcalls?

Reply:
Yeah so I had that originally, but that's where I ran into the `pytest` setup problems, because it would create that `hasher` object during one of the imports, before my monkeypatch to use a mock key vault was applied, so it would try to set up an actual key vault connection with `AzureKeyVault()`, which would then fail because the environment variables are not set... I got really confused as to why the monkeypatch wasn't working as I expected, so ended up just moving these calls inside API calls 🤷‍♂️ 

But agreed that it's not very nice. On the other hand, I'm now actually in the process of removing all the other endpoints except for `/hash`, and moving the special handling logic for patient ID and accession number into `dcmd`, which feels more reasonable?

I'll have another look to see if I can fix the hasher object creation in a way that is compatible with the tests (or vice versa).

Reply:
what are we importing in the `conftest` which ends up importing this module? Should be able to monkeypatch if we're not importing it, and I wouldn't expect the conftest to import much, but could always import locally in a fixture to avoid this?

Reply:
I updated the `Hasher` class slightly. Now it takes the project slug as a parameter during initialisation, so now it has to be created within the API call anyway, I think?

Reply:
Resolving for now, but open to more feedback!